modules/navier_stokes/src/kernels/INSADEnergySUPG.C

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "INSADEnergySUPG.h"
+
+registerMooseObject("NavierStokesApp", INSADEnergySUPG);
+
+InputParameters
+INSADEnergySUPG::validParams()
+{
+  InputParameters params = ADKernelSUPG::validParams();
+  params.addClassDescription("Adds the supg stabilization to the INS temperature/energy equation");
+  params.set<MaterialPropertyName>("tau_name") = "tau_energy";
+  return params;
+}
+
+INSADEnergySUPG::INSADEnergySUPG(const InputParameters & parameters)
+  : ADKernelSUPG(parameters),
+    _temperature_strong_residual(getADMaterialProperty<Real>("temperature_strong_residual"))
+{
+}
+
+ADReal
+INSADEnergySUPG::precomputeQpStrongResidual()
+{
```

Comment:
Why this is not `precomputeQpResidual`? You use `precomputeQpStrongResidual` in other classes.

Reply:
This inherits from a different class called `ADKernelSUPG` which itself inherits from `ADKernelStabilized` which changes the function you have to override to `precomputeQpStrongResidual`